clippy_lints/src/use_self.rs

```diff
@@ -103,6 +103,7 @@ impl<'tcx> LateLintPass<'tcx> for UseSelf {
             if parameters.as_ref().map_or(true, |params| {
                 !params.parenthesized && !params.args.iter().any(|arg| matches!(arg, GenericArg::Lifetime(_)))
             });
+            if !is_from_proc_macro(cx, item); // expensive, should be last check
```

Comment:
Would `item_path.span.from_expansion()` be enough here?

Reply:
Just tried testing it, and `from_expansion` still produces the false positive.

Reply:
huh, what about for `item.span`?

Reply:
The issue is the proc macro using a real span instead of making a new one.

Reply:
The path span happened to do that but the whole item doesn't, so checking its span should be fine

Reply:
It depends what span the proc macro sets. If it doesn't set a span with expansion info, but just uses the span given from the item/tokens it looks to clippy as if it is unexpanded code.